riffle-server/src/store/mem/buffer/unified_buffer.rs

```diff
@@ -0,0 +1,110 @@
+use crate::store::mem::buffer::default_buffer::DefaultMemoryBuffer;
+use crate::store::mem::buffer::opt_buffer::OptStagingMemoryBuffer;
+use crate::store::mem::buffer::unified_buffer::UnifiedBuffer::{DEFAULT, EXPERIMENTAL};
+use crate::store::mem::buffer::{BufferOptions, BufferSpillResult, BufferType, MemoryBuffer};
+use crate::store::{Block, PartitionedMemoryData};
+use croaring::Treemap;
+
+/// this is the router to delegate to the underlying concrete implementation without any cost
+pub enum UnifiedBuffer {
```

Comment:
This is the router buffer to eliminate generic-type leak to the upper structs. And I don't use the `Box<dyn Buffer>` to avoid the dynamic dispatcher cost. FInally I reached here.